### PR TITLE
Fix log printing and ds block number for microblock

### PR DESCRIPTION
**Pull request by bb111189**
## Description

This PR fixes https://github.com/Zilliqa/Issues/issues/89 and https://github.com/Zilliqa/Issues/issues/90. 

1. Fixes ds block number for microblock (https://github.com/Zilliqa/Issues/issues/89)

2. Fixes log printing of variables that are uint8. (https://github.com/Zilliqa/Issues/issues/90)
- difficulty
- block type 

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test
- [ ] small-scale cloud test

### CODE REVIEW

File: src/libNode/MicroBlockProcessing.cpp [hunk 1]
```diff
@@ -762,7 +762,8 @@ bool Node::ComposeMicroBlock()
     TxnHash txRootHash;
     uint32_t numTxs = 0;
     const PubKey& minerPubKey = m_mediator.m_selfKey.second;
-    uint64_t dsBlockNum = m_mediator.m_currentEpochNum;
+    uint64_t dsBlockNum
```
Comment (ansnunez) [inline src/libNode/MicroBlockProcessing.cpp]: Shouldn't it be division, not modulo?

Comment (bb111189) [reply src/libNode/MicroBlockProcessing.cpp]: Indeed. It should be division.

Comment (bb111189) [reply src/libNode/MicroBlockProcessing.cpp]: Fixed.

File: src/libNode/MicroBlockProcessing.cpp [hunk 2]
```diff
@@ -762,7 +762,8 @@ bool Node::ComposeMicroBlock()
...
+        = m_mediator.m_currentEpochNum / NUM_FINAL_BLOCK_PER_POW;
```
Comment (kaikawaliu) [inline src/libNode/MicroBlockProcessing.cpp]: Can just use `m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum()`

**Comment by bb111189** [comment]
Testing in progress

**Comment by bb111189** [comment]
Had rebase to current master and tested this PR on local test. @ansnunez  Ready for a re-review

**Pull request merged**